pdf_build_src/process_markdowns.py

```diff
@@ -141,6 +142,194 @@ def modify_changelog():
+            # Set the column width
+            column_width = max_chars_in_cols[j]
+            if j == 1:
+                column_width = first_column_width
+            elif j == 2:
+                column_width = second_column_width
+            
+            if j == 0 or j == len(row) - 1:
+                row_content.append(elem)
+            else:
+                if '`' in elem:
+                    str_format = ' {:{align}{width}} '
+                    row_content.append(str_format.format(elem, align='<', width=(column_width)))
+                elif '-:' in elem and ':-' in elem :
+                    str_format = ' {:-{align}{width}}: '
+                    row_content.append(str_format.format(':-', align='<', width=(column_width)))
+                elif not '-:' in elem and ':-' in elem :
+                    str_format = ' {:-{align}{width}} '
+                    row_content.append(str_format.format(':-', align='<', width=(column_width)))
+                elif '-:' in elem and not ':-' in elem :
+                    str_format = ' {:-{align}{width}}: '
+                    row_content.append(str_format.format('-', align='<', width=(column_width)))
+                elif i == 1 and not '-:' in elem and not ':-' in elem :
+                    str_format = ' {:-{align}{width}} '
+                    row_content.append(str_format.format('-', align='<', width=(column_width)))
+                else:
+                    row_content.append(str_format.format(elem, align='<', width=(column_width)))
+        if debug:
+            print(row_content)
+
+        new_table.append(row_content)
+        
+    return new_table
+
+
+def correct_tables(root_path):
+    """Change tables in markdown files for correct rendering in PDF.
+
+    This modification makes sure that the proportion and number of dashes (---) are 
+    sufficiently enough for correct PDF rendering and fences (|) are corrected aligned.
+
+    Parameters
+    ----------
+    root_path : str
+        Path to the root directory containing the markdown files
+
+    """
+    markdown_list = []
+    for root, dirs, files in os.walk(root_path):
+        for file in files:
+            if file.endswith(".md") and file != 'index.md' and file != '01-contributors.md' and file != '04-entity-table.md':
+                print('Check tables in {}'.format(os.path.join(root, file)))
+                markdown_list.append(os.path.join(root, file))
+                with open(os.path.join(root, file),'r') as f:
+                    content = f.readlines()
+                tables = []
+                table_mode = False
+                start_line = 0
+                new_content = []
+                for line_nb, line in enumerate(content):
+                    if line:
+                        # Use dashes to detect where a table start and 
+                        # extract the header and the dashes lines
+                        if '--' in line and '|' in line and not table_mode:
```

Comment:
this seems a bit unstable, just imagine we have `--` and `|` somewhere on a line in the spec but it's not a table.

Perhaps the probability of this happening can be decreased by checking `if '---' in line` instead of just `--`?

or counting the number of `|` in that line --> a table would have at least two of those.

an idea would be to replace this line with:

`if _line_contains_table_start(line):`

where 
```python
def _line_contains_table_start(line):
    """Check if line is start of a md table."""
    is_table = False
    
    # do all the checks
    # perhaps set is_table to True if checks pass

    return is_table
```